pmg/views.py

```diff
@@ -208,7 +208,9 @@ def bills(bill_type, year=None):
     api_url = 'bill' if bill_type == 'all' else 'bill/%s' % bill_type
     bills = load_from_api(api_url, return_everything=True, params=params)['results']
 
-    bills.sort(key=lambda b: [-b['year'], b['type']['prefix'], b.get('number', 0), b['title']])
+    # Sort bills by number, date_of_introduction, and then other attributes.
+    bills.sort(key=lambda b: [-b['year'], b['type']['prefix'], b['title']])
```

Comment:
I don't think this will have the desired effect. The second sort will override the first. I suggest:

`bills.sort(key=lambda b: [-b['year'], b['type']['prefix'], b.get('number', 0), b['date_of_introduction'], b['title']])`

This will put the most recent years first, then sort by type, then number, then date (ascending), then break ties (if any) by sorting on the title.

Reply:
I'm not sure if this will work as we need either. It will give results like this:

![screenshot at jul 12 09-59-31](https://user-images.githubusercontent.com/11753432/42620139-6223eca4-85ba-11e8-817a-4edcc09666ea.png)

The way I understood it was that the number and date of introduction should take preference (both in reverse order, to put the latest first, and using both, as bills could have its number or d.o.i returned as None).

Because numbers are sometimes None, `b.get('number', 0)` will return None, and we can't use `-b.get('number', 0)`. We need date of intro in reverse too.

The code I had above seemed to be giving the results we're expecting. Am I misunderstanding the docs I linked to above? In sorting twice to sort some keys in reverse and others in ascending order.